slog/slog_test.go

```diff
@@ -437,3 +437,89 @@ func TestContext(t *testing.T) {
 		t.Errorf("failed to retrieve correct entry from context:\n%s", diff)
 	}
 }
+
+// Examples
+func ExampleNewContext() {
+	entry := WithDetail("id", "123456789")
+	ctx := NewContext(context.Background(), entry)
+	got := FromContext(ctx)
+	fmt.Print(got.Details["id"])
```

Comment:
For these I don't think it makes much sense to `fmt.Print` various internal details, as that isn't how people are going to interact with the libraries. Probably should remove the `// Output:` sections and use `entry.Info("entry pulled from context")`. Where possible we should strive to have examples be functional if copied and pasted into a consumer's code base.

Same comment for all the below instances.